File: versus/dicts/Dicts.java
import java.util.HashMap;

public class Dicts {
    public static void main(String[] args) {
        long n = 3000000, k = 200000;
        HashMap<String, Long> counts = new HashMap<>();
        for (long i = 0; i < n; i++) {
            counts.merge(Long.toString(i % k), 1L, Long::sum);
        }
        long total = 0;
        for (long j = 0; j < k; j++) {
            total += j * counts.get(Long.toString(j));
        }
        System.out.println(total);
    }
}

File: versus/dicts/dicts.rs
use std::collections::HashMap;

fn main() {
    let n: i64 = 3000000;
    let k: i64 = 200000;
    let mut counts: HashMap<String, i64> = HashMap::new();
    for i in 0..n {
        *counts.entry((i % k).to_string()).or_insert(0) += 1;
    }
    let mut total: i64 = 0;
    for j in 0..k {
        total += j * counts[&j.to_string()];
    }
    println!("{}", total);
}
